mlir/lib/Dialect/Bufferization/Transforms/BufferResultsToOutParams.cpp

```diff
@@ -43,6 +45,49 @@ static bool hasStaticIdentityLayout(MemRefType type) {
   return type.getLayout().isIdentity();
 }
 
+/// Return the dynamic shapes of the `memref` based on the defining op. If the
+/// complete dynamic shape fails to be captured, return an empty value.
+/// Currently, only function block arguments are supported for capturing.
+static SmallVector<Value> getDynamicSize(Value memref, func::FuncOp funcOp) {
+  Operation *defOp = memref.getDefiningOp();
+  if (!defOp)
+    return {};
+  auto operands = defOp->getOperands();
+  SmallVector<Value> dynamicSizes;
+  for (Value size : operands) {
+    BlockArgument sizeSrc = dyn_cast<BlockArgument>(size);
+    if (!sizeSrc)
+      return {};
+
+    auto iter = llvm::find(funcOp.getArguments(), sizeSrc);
+    if (!iter)
```

Comment:
This may be fine, but I think I've seen `iter == funcOp.getArguments().end()` in the past. Please double check that `!iter` actually works before merging.